src/signed_stubs.c

```diff
@@ -0,0 +1,143 @@
+    for (;; pos++) {                                                         \
+      if (*pos == '_') continue;                                             \
+      d = parse_digit(*pos);                                                 \
+      /* Halt if the digit isn't valid (or this is the string terminator) */ \
+      if (d < 0 || d >= base) break;                                         \
+      /* Check that we can add another digit */                              \
+      if (i > max_prefix) break;                                             \
+      i = (sign < 0 ? (- d) : d) + i * base;                                 \
+      /* Check for overflow */                                               \
+      if (sign < 0) {                                                        \
+        if (i > ((TYPE(BITS)) (- d))) break;                                 \
+      } else {                                                               \
+        if (i < (TYPE(BITS)) d) break;                                       \
+      }                                                                      \
+    }                                                                        \
+                                                                             \
+    if (pos != String_val(a) + caml_string_length(a)){                       \
+      caml_failwith("Int"#BITS".of_string");                                 \
+    }                                                                        \
+                                                                             \
+    return COPY(i);                                                          \
+  }                                                                          \
+
+#define INT_SMALL_DEFS(BITS)                                                 \
+  /* of_string : string -> t */                                              \
+  INT_OF_STRING(BITS, Integers_val_int ## BITS)                              \
+                                                                             \
+  /* to_string : t -> string */                                              \
+  value integers_int ## BITS ## _to_string(value a)                          \
+  {                                                                          \
+    char buf[BUF_SIZE(TYPE(BITS))];                                          \
+    if (sprintf(buf, "%" PRId ## BITS , Int ## BITS ##_val(a)) < 0)          \
+      caml_failwith("Int ## BITS ## .to_string");                            \
+    else                                                                     \
+      return caml_copy_string(buf);                                          \
+  }                                                                          \
+                                                                             \
+  /* to_hexstring : t -> string */                                           \
```

Comment:
The specification of `to_hexstring` (which should really be more clearly documented in the existing code) is 

> hex pretty-printing (to mimic "%x") 

(see https://github.com/yallop/ocaml-integers/issues/33#issuecomment-881456760).

which means, e.g.:

```ocaml
# Printf.sprintf "%x" (-15);;
- : string = "7ffffffffffffff1"
```